lib/matplotlib/cbook/__init__.py

```diff
@@ -2766,3 +2766,19 @@ def _topmost_artist(
     in reverse order.
     """
     return _cached_max(reversed(artists))
+
+
+def _str_equal(obj, s):
+    """Return whether *obj* is a string equal to string *s*.
+
+    This helper solely exists to handle the case where *obj* is a numpy array.
```

Comment:
I find this confusing.  *How* does it handle that case?  It returns False, so that `np.array('abc')` is not equal to 'abc'.  Where is this needed? And why is it only the numpy array case that needs to be handled?  I can imagine there might be other places in the code where this might be useful because 'obj' might be None, or a number, or...

Reply:
if obj is a number or None or etc., then, `obj == s` will return False just fine.  The problem with ndarrays is that `np.array(...) == s` will return an array, which cannot be used in a boolean switch context.

The fact that `np.array("abc")` will compare unequal to "abc" did not change with this PR, we were previously already using isinstance().

Perhaps the docstring can be clarified to
> Return whether *obj* is a string equal to string *s*.
>
> This helper solely exists to handle the case where *obj* is a numpy array, because in such cases, a naive `obj == s` would yield an array, which cannot be used in a boolean context.

How does that sound?  Feel free to propose further edits.

Reply:
That's good, thank you.

Reply:
edited accordingly (both instances)